### DISCUSSION TITLE
Borg and Unique Hunting

### DISCUSSION THREAD

**Discussion by Netbrian** [angband/angband#6077] [Ideas]
@agoodman00 

When the borg gets to level 98 and decides to kill every unique, does he have any particular strategy, or does he just wander around dlvl 98 until everyone shows up?

Right now my borg is searching for the Mouth of Sauron, and having a terrible time.

**Comment by agoodman00** [comment]
Basically the code is (pseudo code)
if max depth is 99, gather up potions to go after morgoth
if in dungeon and have potions, check if there are uniques to kill
true
   go to l98 and hope to run into uniques 
false
  go to 100.  

Since he has the potions that he is going to use to go after morgoth, he should have a slightly easier time killing other uniques.  Often it seems to take a while.  I think the strategy could use improvement but there have always been more high priority issues to work on.

**Comment by Netbrian** [comment]
I noticed the following scenario, which isn't quite a bug --

1.  The borg loses stats, and goes to shallow levels to find restore potions.
2. The borg drinks the potion and restores his stats.
3. The borg stays at the shallow levels and doesn't think to go back to town so he can return to dlvl 98.  He just putters around until he's forced to recall for some other reason.

**Comment by agoodman00** [comment]
Yup.  That is normal borg behavior.  The only check he does is "am I way too shallow?  Dive deeper".  So he shouldn't be hanging around the shallow levels but diving deep as fast as he has a level feeling and can find stairs.... as long as there isn't reason to hang out in a level, like it having a good feeling.  But he is just walking down.

**Reply by Netbrian** [reply]
Would it be possible to tweak this to encourage him to recall back to town if his current depth is over ten levels shallower than the "target" depth?

**Reply by agoodman00** [reply]
hmmm.  I think the best place to do that would be when he is checking if he should leave the level.  I think we would want an added condition for "recall depth not much deeper than target (safe) depth".  
I don't think that would be hard to add.  Trying to sort out the stat issue right now.  The code for keeping track of stats is just... not right.

**Reply by agoodman00** [reply]
of course if you wanted to hop in and add the code ....  :-D

**Reply by Netbrian** [reply]
I've been looking into it, but I'm soft and used to more modern languages.  All these fancy context variables scare me.

**Reply by agoodman00** [reply]
let me encourage you.  I certainly don't mind checking over your code as you get used to C.
